### PR TITLE
Polish declarative merge documentation grammar

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- clarify the MergeLayer description in `docs/design.md` with improved grammar and wrapping for readability
- fix user guide sentences that join independent clauses with "so" by adding the required commas in both affected sections

## Testing
- make fmt
- make check-fmt
- make lint
- make test
- make markdownlint
- make nixie

------
https://chatgpt.com/codex/tasks/task_e_68f4384479908322a3f664e4a42bed86

## Summary by Sourcery

Enable field-level append semantics in the declarative merge derive macro so Vec fields accumulate JSON values across layers, and update documentation and examples to reflect and clarify the new behavior.

New Features:
- Support append merge strategy for Vec fields in declarative merge derive macro

Enhancements:
- Collect and propagate append_fields into generated merge state and logic to accumulate vector values across layers

Documentation:
- Revise design document, user guide, examples, and roadmap to describe the vector append strategy and improve grammar

Tests:
- Add unit tests for append field code generation and integration tests for vector append and optional null merge behaviors

### CODE REVIEW

File: ortho_config_macros/src/lib.rs [hunk 1]
```diff
@@ -398,25 +400,78 @@ fn generate_ortho_impl(
     }
 }
 
-fn generate_declarative_state_struct(state_ident: &syn::Ident) -> proc_macro2::TokenStream {
+fn generate_declarative_state_struct(
```
[the remark this answers is not in the document]
Comment (leynos) [reply ortho_config_macros/src/lib.rs]: Has this now been resolved?

**Comment by leynos** [comment]
@coderabbitai Please suggest a fix for this issue and supply a prompt for an AI coding agent to enable it to apply the fix:

[ortho_config_macros/src/derive/generate/declarative/tests.rs](https://github.com/leynos/ortho-config/pull/210/files/6623ff2bd0ff0c41d7d815b349ced3fbbe83a3ee#diff-6a29e91ed92913b955d4eae349c82ae26267cf6a5a6f9d578863639d64abe82a)

Comment on lines +100 to +174

~~~rust
fn generate_declarative_merge_impl_emits_trait_impl() {
    let state_ident = parse_str("__SampleDeclarativeMergeState").expect("state ident");
    let config_ident = parse_str("Sample").expect("config ident");
    let tokens = generate_declarative_merge_impl(&state_ident, &config_ident, &[]);
    let expected = quote! {
        impl ortho_config::DeclarativeMerge for __SampleDeclarativeMergeState {
            type Output = Sample;

            fn merge_layer(
                &mut self,
                layer: ortho_config::MergeLayer<'_>
            ) -> ortho_config::OrthoResult<()> {
                use ortho_config::OrthoResultExt as _;

                let provenance = layer.provenance();
                let path = layer.path().map(|p| p.to_owned());
                let value = layer.into_value();
                let mut map = match value {
                    ortho_config::serde_json::Value::Object(map) => map,
                    other => {
                        let provenance_label = match provenance {
                            ortho_config::MergeProvenance::Defaults => "defaults",
                            ortho_config::MergeProvenance::File => "file",
                            ortho_config::MergeProvenance::Environment => "environment",
                            ortho_config::MergeProvenance::Cli => "CLI",
                            _ => "unknown",
                        };
                        let value_kind = match other {
                            ortho_config::serde_json::Value::Null => "null",
                            ortho_config::serde_json::Value::Bool(_) => "a boolean",
                            ortho_config::serde_json::Value::Number(_) => "a number",
                            ortho_config::serde_json::Value::String(_) => "a string",
                            ortho_config::serde_json::Value::Array(_) => "an array",
                            ortho_config::serde_json::Value::Object(_) => unreachable!(
                                "objects handled by earlier match arm"
                            ),
                        };
                        let mut message = format!(
                            concat!(
                                "Declarative merge for ",
                                stringify!(Sample),
                                " expects JSON objects but the ",
                                "{provenance_label} layer supplied {value_kind}. "
                            ),
                            provenance_label = provenance_label,
                            value_kind = value_kind,
                        );
                        if let Some(path) = path {
                            message.push_str("Source: ");
                            message.push_str(path.as_str());
                            message.push_str(". ");
                        }
                        message.push_str("Non-object layers would overwrite accumulated state.");
                        return Err(std::sync::Arc::new(ortho_config::OrthoError::merge(
                            ortho_config::figment::Error::from(message),
                        )));
                    }
                };
                if !map.is_empty() {
                    ortho_config::declarative::merge_value(
                        &mut self.value,
                        ortho_config::serde_json::Value::Object(map),
                    );
                }
                Ok(())
            }

            fn finish(self) -> ortho_config::OrthoResult<Self::Output> {
                let __SampleDeclarativeMergeState { mut value, } = self;
                ortho_config::declarative::from_value(value)
            }
        }
    };
    assert_eq!(tokens.to_string(), expected.to_string());
}
~~~

❌ New issue: [Large Method](https://codescene.io/projects/69279/delta?repo-id=1160464&review-id=210&biomarker=Large+Method&filename=ortho_config_macros%2Fsrc%2Fderive%2Fgenerate%2Fdeclarative%2Ftests.rs&method=generate_declarative_merge_impl_emits_trait_impl)
generate_declarative_merge_impl_emits_trait_impl has 72 lines, threshold = 70
